akka-http/src/main/scala/akka/http/scaladsl/server/Rejection.scala

```diff
@@ -92,6 +93,15 @@ final case class MalformedHeaderRejection(headerName: String, errorMsg: String,
   extends jserver.MalformedHeaderRejection with RejectionWithOptionalCause
 
 /**
+ * Rejection created by [[akka.http.scaladsl.server.directives.HeaderDirectives.checkSameOrigin]].
+ * Signals that the request was rejected because `Origin` header value is invalid.
+ */
+final case class InvalidOriginHeaderRejection(invalidOrigins: immutable.Seq[HttpOrigin])
+  extends jserver.InvalidOriginHeaderRejection with Rejection {
+  override def getInvalidOrigins: java.util.List[HttpOrigin] = invalidOrigins.asJava
```

Comment:
so `invalidOrigins` has to be `immutable.Seq[akka.http.scaladsl.model.headers.HttpOrigin]`
and this has to be `java.util.List[javadsl...HttpOrigin]`, so we'll need to do `.map(_.asJava).asJava` I fear...
We may get away with a cast, but let's not try for now